docs/TIPS.md

```diff
@@ -0,0 +1,251 @@
+
+Adding `-p -q` to `./pyston_release` can output some more information about jitted frames that perf can use. Perf has a few other useful flags, such as `-e page-faults` which counts the number of page faults during an execution. While we optimize some of it away, Python intrinsically allocates massive amounts of memory and we are definitely at the point where it is worth thinking about caching, at all levels of abstraction.
+
+For some notes on other profiling tools, see [PROFILING](PROFILING).
+
+## Travis CI
+
+We use Travis CI for continuous integration. Pull requests to the main repository will be run automatically against our full test suite upon creation and updates.
+
+##### Problem: My code passes all the tests locally but fails on Travis CI
+/spp
+Travis CI runs the tests in release mode, so you may want to try `make check_release` or `make check_release ARGS=TESTNAME` in your local machine. If the Clang build works but not the GCC build, build with GCC using `make pyston_gcc`.
+
+Travis CI runs on Ubuntu 12.04.5 LTS - having a similar setup would definitely help.
+
+## Debugging
+
+When Pyston crashes, the first thing you want to do is to get the crash location inside GDB. How do you continue from that point on?
+
+Looking at the classes of objects in the current stack frame is always a good start. Any C++ type that inherits from `Box` (of the form `Box.*`) should have a `cls` field which stores a `BoxedClass`, the class of the object. You would want to use `p obj->cls->tp_name`. If the value looks like garbage, there is a good chance that the object has either been corrupted or that we can somehow access memory that has been freed. Although you won't see them directly in the C++ definition of `BoxedClass`, `BoxedClass`s have [all the fields that type objects in CPython do](https://docs.python.org/2/c-api/typeobj.html) (fields with the prefix `tp_`).
+
+In general, boxes are allocated in the heap (but not all - e.g. classes of built-in types) in one of the three heaps. Pointers in the heap have a hardcoded range of allowed values (see `src/gc/heap.h`) - values that fall outside this range (and are not in the stack) could potentially indicate memory errors.
+
+The C++ stacktrace can be quite difficult to read. Every Python function call will result in half a dozen C++ function calls along the lines of `getAttrInternal`. This is why the `_printStacktrace()` function, which can be used inside GDB, can come in very handy.
+
+##### Problem: The tests fail in release mode
+
+For example, `tests/sys_argv` might output `['/mnt/rudi/pyston/test/tests/sys_argv.py']` instead of `['/mnt/rudi/pyston/./test/tests/sys_argv.py']`. If you switch between testing in debug mode and release mode, try running `touch test/tester.py` in-between runs.
```

Comment:
I think this is due to switching between the cmake-based test targets (check and quick_check I believe) and the makefile-based ones (check_dbg, check_release).